src/main/java/view/MainMenu.java

```diff
@@ -34,17 +39,40 @@ private MainMenu(JPanel cur){
         //each semester, then add CourseNodes for each class in that semester
         // Root is used to keep semesters together
         DefaultMutableTreeNode root = new DefaultMutableTreeNode("Root", true);
-        DefaultMutableTreeNode semester = new DefaultMutableTreeNode("2018 Spring", true);
-        CourseNode node1 = new CourseNode("cs591 d1", semester.toString(), getNextCourseId());
-        node1.addStudents(Student.loadStudentsFromFile(new File("./src/main/java/view/stu.txt")));
+
+        java.util.List<ClassModel> classes = loadClasses();
+        java.util.List<Semester> semesters = loadSemesters();
+        Map<Integer, SemesterNode> semesterMap = new HashMap<>(); //Maps semesters to their SemesterNode
+
+        for(Semester sem: semesters) {
+            SemesterNode semester = new SemesterNode(sem.getName(), true, sem.getId());
+            System.out.println(sem.getName());
+            root.add(semester);
+            semesterMap.put(sem.getId(), semester);
+        }
+        System.out.println(semesterMap);
+        for(ClassModel clazz: classes) {
+            if(semesterMap.containsKey(clazz.SemesterID)) {
+                System.out.println("Adding to semester node: "+clazz.CourseNumber);
+                SemesterNode semester = semesterMap.get(clazz.SemesterID);
+                CourseNode course = new CourseNode(clazz, semester.getSemesterName());
+                semester.add(course);
+            } else {
+                System.out.println("Adding to root node: "+clazz.CourseNumber);
+                CourseNode course = new CourseNode(clazz, "");
```

Comment:
Since each class should be part of a semester, this shouldn't happen. If it does happen, maybe we can create a Semester and SemesterNode on the fly?